src/main/java/analyzer/analyzerInterfaces/AnalyzerComment.java

```diff
@@ -0,0 +1,22 @@
+package analyzer.analyzerInterfaces;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+public interface AnalyzerComment {
+
+    // Constructed message with all the template variables applied.
+    String message();
+
+    // The message with template variables in there.
+    String template();
+
+    // The identifier for the template on website-copy.
+    String externalTemplate();
+
+    // Not sure if this should be a map
+    ArrayList<String> variables();
+
+    // The type of comment.
+    Optional<?> commentType();
```

Comment:
I assume the `?` is temporary? You can find the comment types here: https://exercism.org/docs/building/tooling/analyzers/interface#h-type-optional

Reply:
Well the ? means "generic type" because I'm just not sure what should go here. Looking at how the JS analyzer is set up, it is similar. I'm aware of the different comment types, I'm just trying to figure out the most appropriate way to set up this interface to have multiple comment types. I will work on this and push again. @ErikSchierboom

Reply:
Could the comment type just be a regular enum?